### PR TITLE
cloud: migrate data from on-premises tidb to tidb cloud

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Add a new article abot how to migrate data from on-premise TiDB to TiDB Cloud

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/migrate-from-op-tidb.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+# Migrate from On-Premise TiDB to TiDB Cloud
+
+本文主要针对 OP(On Premise) 环境自建 TiDB 集群，通过 dumpling + TiCDC 工具，将数据全量导入TiDB Cloud (AWS) 的场景。
+
+整体方案：
+  - 环境搭建及工具准备
+  - 存量数据迁移：OP TiDB ( dumpling ) -- S3 -- (Import) TiDB Cloud 
+  - 增量数据迁移：OP TiDB  -- (TiCDC) -- TiDB Cloud 
+  - 数据验证
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-op-tidb.md:14]:
```suggestion
---
title: Migrate from On-Premise TiDB to TiDB Cloud
summary: Learn how to migrate data from on-premise TiDB to TiDB Cloud.
---

# Migrate from On-Premises TiDB to TiDB Cloud

This document describes how to migrate data from your on-premises (OP) TiDB clusters to TiDB Cloud (AWS) through Dumpling and TiCDC.

The overall procedure is as follows:

- Build the environment and prepare the tools
- Full data migration: OP TiDB (Dumpling) --> S3 --> (Import) TiDB Cloud
- Incremental data migration: OP TiDB --> (TiCDC) --> TiDB Cloud
- Data verification

## Prerequisites

It is recommended that you put the S3 bucket and TiDB Cloud in the same region. Cross-region migration might incur additional data conversion cost.

  1. Prepare an [AWS account](https://docs.aws.amazon.com/AmazonS3/latest/userguide/setting-up-s3.html#sign-up-for-aws-gsg) with administrator access
  2. Prepare an [AWS S3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html)
  3. [Deploy an on-premises TiDB cluster](https://docs.pingcap.com/zh/tidb/dev/hardware-and-software-requirements)
  4. Prepare [a TiDB Cloud account with administrator access and create a TiDB Cloud (AWS) cluster](/tidb-cloud/tidb-cloud-quickstart.md)

## Prepare tools

### Dumpling

[Dumpling](https://docs.pingcap.com/tidb/dev/dumpling-overview) is a data export tool that exports data from TiDB/MySQL into SQL or CSV files. You can use Dumpling to export full data from TiDB Cloud.

Before you deploy Dumpling, note the following:

- It is recommended to deploy Dumpling on a new EC2 instance in the same VPC as the TiDB cluster.
- The EC2 instance should have at least 16 cores and 32 GB of memory to guarantee a good export performance.
- The recommended EC2 instance type is c6g.4xlarge. You can choose other EC2 instance types based on your needs. The AMI can be Amazon Linux, Ubuntu, or Red Hat.

You can deploy Dumpling by using TiUP or using the binary package.

#### Deploy Dumpling usig TiUP

Use [TiUP](https://docs.pingcap.com/tidb/stable/tiup-overview) to deploy Dumpling:

```bash
## Deploy tiup
curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
source /root/.bash_profile
## Deploy Dumpling and update to the latest version
tiup install dumpling
tiup update --self && tiup update dumpling
```

#### Deploy Dumpling using the installation package

To deploy Dumpling using the installation package

1. Download the [toolkit package](https://docs.pingcap.com/tidb/stable/download-ecosystem-tools) from the [Dumpling download page].
2. Extract it to the target machine and [install Dumpling](https://docs.pingcap.com/tidb/stable/dumpling-overview#dumpling-introduction).

#### Configure privileges for Dumpling

You need the following previleges to export data from the upstream database:

- SELECT
- RELOAD
- LOCK TABLES
- REPLICATION CLIENT
- PROCESS
```

**Review by snowballbear — APPROVED** [review]

File: tidb-cloud/migrate-from-op-tidb.md [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    ```shell
+    tiup cdc cli changefeed create \
+    --pd=http://172.16.6.122:2379  \
+    --sink-uri="tidb://root:123456@172.16.6.125:4000"  \
+    --changefeed-id="upstream-to-downstream"  \
+    --start-ts="431434047157698561"
+    ```
+
+    - `--pd`: the PD address of the upstream cluster. The format is: `[upstream_pd_ip]:[pd_port]`
+    - `--sink-uri`: the downstream address of the replication task. Configure `--sink-uri` according to the following format. Currently, the scheme supports `mysql/tidb/kafka/pulsar/s3/local`.
```
[the remark this answers is not in the document]
Comment (shichun-0415) [reply tidb-cloud/migrate-from-op-tidb.md:296]: pulsar 在 6.3.0 移除了：https://github.com/pingcap/docs-cn/pull/11112/files

File: tidb-cloud/migrate-from-op-tidb.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+
+        ```shell
+        [scheme]://[userinfo@][host]:[port][/path]?[query_parameters]
+        ```
+
+    - `--changefeed-id`: the ID of the replication task. The format must match the ^[a-zA-Z0-9]+(\-[a-zA-Z0-9]+)*$ regular expression. If this ID is not specified, TiCDC automatically generates a UUID (the version 4 format) as the ID.
+    - `--start-ts`: specifies the starting TSO of the changefeed. From this TSO, the TiCDC cluster starts pulling data. The default value is the current time.
+
+    For more information, see [Manage TiCDC Cluster and Replication Tasks](https://docs.pingcap.com/tidb/stable/manage-ticdc).
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:305]: ticdc 最近重构了，或许这个链接 https://docs.pingcap.com/tidb/dev/ticdc-changefeed-config 会更合适一点，manage-ticdc 内容比较杂。

File: tidb-cloud/migrate-from-op-tidb.md [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+5. Reopen the GC mechanism in the upstream cluster. After checking the incremental replication, and there is no error or delay in replication, enable the GC mechanism to resume the garbage collection function of the cluster.
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:307]:
```suggestion
5. Enable the GC mechanism again in the upstream cluster. If no error or delay is found in incremental replication, enable the GC mechanism to resume the garbage collection function of the cluster.
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+    ```sql
+    mysql > SET GLOBAL tidb_gc_enable = TRUE;
+    ## Verify whether the setting works. 1 indicates that the GC mechanism is enabled.
+    mysql > SELECT @@global.tidb_gc_enable;
+    +-------------------------+
+    | @@global.tidb_gc_enable |
+    +-------------------------+
+    |                       1 |
+    +-------------------------+
+    1 row in set (0.01 sec)
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:318]: separate the command and output, and remove the `mysql` prompt.

File: tidb-cloud/migrate-from-op-tidb.md [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+    ```
+
+6. Verify the incremental replication task.
+
+    - If the replication task is created successfully, the message "Create changefeed successfully!" is displayed in the output.
+    - Check the status of the replication task. If the state shows `normal`, it means the replication task is normal.
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:324]:
```suggestion
    - If the message "Create changefeed successfully!" is displayed in the output, the replication task is created successfully.
    - If the state is `normal`, the replication task is normal.
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+- RELOAD
+- LOCK TABLES
+- REPLICATION CLIENT
+- PROCESS
+
+### Deploy TiCDC
+
+You need to [deploy TiCDC](https://docs.pingcap.com/tidb/dev/deploy-ticdc) to replicate incremental data from the upstream TiDB cluster to TiDB Cloud.
+
+1. Confirm whether the current TiDB version supports TiCDC. TiDB v4.0.8.rc.1 and later versions support TiCDC. You can check the TiDB version by executing `select tidb_version();` in the TiDB cluster. If you need to upgrade it, see [Upgrade TiDB Using TiUP](https://docs.pingcap.com/tidb/dev/upgrade-tidb-using-tiup).
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-op-tidb.md:82]:
```suggestion
1. Confirm whether the current TiDB version supports TiCDC. TiDB v4.0.8.rc.1 and later versions support TiCDC. You can check the TiDB version by executing `select tidb_version();` in the TiDB cluster. If you need to upgrade it, see [Upgrade TiDB Using TiUP](https://docs.pingcap.com/tidb/dev/deploy-ticdc#upgrade-ticdc-using-tiup).
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+
+2. Add the TiCDC component to the TiDB cluster. See [Scale a TiDB Cluster Using TiUP](https://docs.pingcap.com/tidb/dev/scale-tidb-using-tiup).
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-op-tidb.md:84]:
```suggestion
2. Add the TiCDC component to the TiDB cluster. See [Add or scale out TiCDC to an existing TiDB cluster using TiUP](https://docs.pingcap.com/tidb/dev/deploy-ticdc#add-or-scale-out-ticdc-to-an-existing-tidb-cluster-using-tiup).
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+    - `--sink-uri`: the downstream address of the replication task. Configure `--sink-uri` according to the following format. Currently, the scheme supports `mysql`, `tidb`, `kafka`, `s3`, and `local`.
+
+        ```shell
+        [scheme]://[userinfo@][host]:[port][/path]?[query_parameters]
+        ```
+
+    - `--changefeed-id`: the ID of the replication task. The format must match the ^[a-zA-Z0-9]+(\-[a-zA-Z0-9]+)*$ regular expression. If this ID is not specified, TiCDC automatically generates a UUID (the version 4 format) as the ID.
+    - `--start-ts`: specifies the starting TSO of the changefeed. From this TSO, the TiCDC cluster starts pulling data. The default value is the current time.
+
+    For more information, see [Manage TiCDC Cluster and Replication Tasks](https://docs.pingcap.com/tidb/stable/manage-ticdc).
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-op-tidb.md:306]:
```suggestion
    For more information, see [CLI and Configuration Parameters of TiCDC Changefeeds](https://docs.pingcap.com/tidb/dev/ticdc-changefeed-config).
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 10]
```diff
@@ -0,0 +1,346 @@
...
+#### Step 1. Disable the GC mechanism of the upstream OP TiDB cluster temporarily
+
+To ensure that newly written data is not lost during incremental migration, you need to disable the upstream cluster's garbage collection (GC) mechanism before starting the migration to prevent the system from cleaning up historical data.
+
+Run the following command to verify whether the setting is successful.
+
+```sql
+SET GLOBAL tidb_gc_enable = FALSE;
+
+```
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:127]:
```suggestion
```sql
SET GLOBAL tidb_gc_enable = FALSE;
```
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 11]
```diff
@@ -0,0 +1,332 @@
...
+    - `{schema}-schema-create.sql`: the SQL file for creating the schema
+    - `{schema}.{table}-schema.sql`: the SQL file for creating the table
+    - `{schema}.{table}.{0001}.{sql|csv}`: data files
+    - `*-schema-view.sql`, `*-schema-trigger.sql`, `*-schema-post.sql`: other exported SQL files
+
+### Migrate data from Amazon S3 to TiDB Cloud
+
+After you export data from the on-premises TiDB cluster to Amazon S3, you need to migrate the data to TiDB Cloud.
+
+If your TiDB cluster is in a local IDC, or the network between the Dumpling server and Amazon S3 is not connected, you can export the files to the local storage first, and then upload them to Amazon S3 later.
```
[the remark this answers is not in the document]
Comment (hfxsd) [reply tidb-cloud/migrate-from-op-tidb.md:207]: 好的，和前面重复了。删掉。

File: tidb-cloud/migrate-from-op-tidb.md [hunk 12]
```diff
@@ -0,0 +1,344 @@
...
+
+## Prerequisites
+
+It is recommended that you put the S3 bucket and the TiDB Cloud cluster in the same region. Cross-region migration might incur additional cost for data conversion.
+
+Before migration, you need to prepare the following:
+
+- An [AWS account](https://docs.aws.amazon.com/AmazonS3/latest/userguide/setting-up-s3.html#sign-up-for-aws-gsg) with administrator access
+- An [AWS S3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html)
+- [A TiDB Cloud account with the administrator access and create a TiDB Cloud (AWS) cluster](/tidb-cloud/tidb-cloud-quickstart.md)
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:27]:
```suggestion
- [A TiDB Cloud account with the administrator access and a TiDB Cloud (AWS) cluster](/tidb-cloud/tidb-cloud-quickstart.md)
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 13]
```diff
@@ -0,0 +1,344 @@
...
+# Migrate from On-Premises TiDB to TiDB Cloud
+
+This document describes how to migrate data from your on-premises (OP) TiDB clusters to TiDB Cloud (AWS) through Dumpling and TiCDC.
+
+The overall procedure is as follows:
+
+1. Build the environment and prepare the tools.
+2. Migrate full data. The process is as follows:
+   1. Export data from OP TiDB to Amazon S3 using Dumpling.
+   2. Import from Amazon S3 to TiDB Cloud.
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:15]:
```suggestion
   2. Import data from Amazon S3 to TiDB Cloud.
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 14]
```diff
@@ -0,0 +1,344 @@
...
+    2. Add the TiCDC component and check the status.
+
+      ```shell
+      tiup cluster scale-out <cluster-name> scale-out.yaml
+      tiup cluster display <cluster-name>
+      ```
+
+## Migrate full data
+
+To migrate data from the on-premises TiDB cluster to TiDB Cloud, perform a full data migration as follows:
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:107]:
```suggestion
To migrate data from the OP TiDB cluster to TiDB Cloud, perform a full data migration as follows:
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 15]
```diff
@@ -0,0 +1,344 @@
...
+
+1. Migrate data from the on-premises TiDB cluster to Amazon S3.
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:109]:
```suggestion
1. Migrate data from the OP TiDB cluster to Amazon S3.
```

File: tidb-cloud/migrate-from-op-tidb.md [hunk 16]
```diff
@@ -0,0 +1,344 @@
...
+
+5. Enable the GC mechanism again in the upstream cluster. If no error or delay is found in incremental replication, enable the GC mechanism to resume garbage collection of the cluster.
+
+    Run the following command to verify whether the setting works.
+
+    ```sql
+    SET GLOBAL tidb_gc_enable = TRUE;
+    ```
+
+    The following is an example output, in which `1` indicates that it is disabled.
```
Comment (shichun-0415) [inline tidb-cloud/migrate-from-op-tidb.md:321]:
```suggestion
    The following is an example output, in which `1` indicates that GC is disabled.
```

**Review by shichun-0415 — APPROVED** [review]

**Review by shichun-0415 — APPROVED** [review]

**Review by Frank945946 — APPROVED** [review]

File: TOC-tidb-cloud.md [hunk 17]
```diff
@@ -96,24 +96,20 @@
   - [Upgrade a TiDB Cluster](/tidb-cloud/upgrade-tidb-cluster.md)
   - [Delete a TiDB Cluster](/tidb-cloud/delete-tidb-cluster.md)
 - Migrate Data
```
Comment (hfxsd) [inline TOC-tidb-cloud.md:98]:
```suggestion
- Migrate or Import Data
```

**Pull request merged**